tests/e2e/fullsync_test_for_block_volume.go

```diff
@@ -769,4 +770,114 @@ var _ bool = ginkgo.Describe("full-sync-test", func() {
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
 
+	/*
+		Attach volume to a new pod when CNS is down and verify volume metadata in CNS post full sync
+		Steps:
+			1.	create a pvc pvc1, wait for pvc bound to pv
+			2.	create a pod pod1, using pvc1
+			3.	stop vsan-health on VC
+			4.	when vsan-health is stopped, delete pod1
+			5.	create a new pod pod2, using pvc1
+			6.	start vsan-health on VC
+			7.	after fullsync is triggered verify the CNS metadata for the volume backing pvc1
+			8.	delete pod2
+			9.	delete pvc1
+	*/
+	ginkgo.It("[csi-block-vanilla] [csi-supervisor] [csi-guest] [csi-block-vanilla-serialized] "+
+		"Attach volume to a new pod when CNS is down and verify volume metadata in CNS post full sync", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("create a pvc pvc1, wait for pvc bound to pv")
+		volHandle, pvc, pv, _ := createSCwithVolumeExpansionTrueAndDynamicPVC(
+			f, client, "", storagePolicyName, namespace)
+
+		if guestCluster {
+			volHandle = getVolumeIDFromSupervisorCluster(volHandle)
+			gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		}
+
+		ginkgo.By("create a pod pod1, using pvc1")
+		pod, _ := createPODandVerifyVolumeMount(f, client, namespace, pvc, volHandle)
+
+		ginkgo.By("Stopping vsan-health on the vCenter host")
+		vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+		isVsanhealthServiceStopped = true
+		err := invokeVCenterServiceControl(stopOperation, vsanhealthServiceName, vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow vsan-health to completely shutdown",
+			vsanHealthServiceWaitTime))
+		time.Sleep(time.Duration(vsanHealthServiceWaitTime) * time.Second)
```

Comment:
Add a todo to replace this logic with waitVCenterServiceToBeInState from PR 
https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1246/files#diff-9b38283911a9007988741db3d12b060b40588df5ad95147e0bf156f35152a9baR814

Reply:
Added